firestore/integration_test_internal/src/android/arena_ref_android_test.cc

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "firestore/src/jni/arena_ref.h"
+
+#include "firestore_integration_test_android.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+namespace firebase {
+namespace firestore {
+namespace jni {
+namespace {
+
+using ArenaRefTestAndroid = FirestoreAndroidIntegrationTest;
+
+TEST_F(ArenaRefTestAndroid, DefaultConstructorCreatesReferenceToNull) {
+  Env env;
+  ArenaRef arena_ref;
+  EXPECT_EQ(arena_ref.get(env).get(), nullptr);
+}
+
+TEST_F(ArenaRefTestAndroid, ConstructFromEnvAndObject) {
+  Env env;
+  Local<String> string = env.NewStringUtf("hello world");
+
+  ArenaRef arena_ref(env, string);
+  EXPECT_TRUE(arena_ref.get(env).Equals(env, string));
+}
+
+TEST_F(ArenaRefTestAndroid, CopyConstructor) {
```

Comment:
I have a question: Both of them invokes copy ctor function, IMO they are testing one thing.

Reply:
I disagree. They literally invoke the exact same constructor so there is no reason to test both of them. By testing both of them you are essentially testing the C++ compiler to make sure it calls the right constructor. The tests are supposed to be testing _your_ code, not the compiler. If you feel strongly that you'd like to leave in both ways of invoking the copy constructor then please just add a comment to explicitly state what you are doing. Otherwise, to me,  it was confusing because it looked like there was some confusion about how C++ works, which is not the case.